src/statement-list/statement_list_parse_tree_io.cpp

```diff
@@ -190,8 +210,21 @@ void output_instruction(
     {
       if(expr.id() == ID_symbol)
         out << '\t' << expr.get(ID_identifier);
-      else if(expr.id() == ID_constant)
-        out << '\t' << expr.get(ID_value);
+      else if(can_cast_expr<constant_exprt>(expr))
+      {
+        const constant_exprt &constant = to_constant_expr(expr);
+        mp_integer ivalue;
```

Comment:
The conversion of the value is duplicated. Extract to a `static` function which accepts an `std::ostream &` to only write the resulting value to the stream, e.g.:
```
out << '\t';
print(out, constant);
```